pkg/controllers/updaterun/controller.go

```diff
@@ -312,6 +317,34 @@ func handleClusterApprovalRequest(oldObj, newObj client.Object, q workqueue.Type
 	})
 }
 
+// handleClusterApprovalRequestDelete finds the ClusterStagedUpdateRun creating the ClusterApprovalRequest,
+// and enqueues it to the ClusterStagedUpdateRun controller queue when the ClusterApprovalRequest is deleted.
+func handleClusterApprovalRequestDelete(obj client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	appReq, ok := obj.(*placementv1beta1.ClusterApprovalRequest)
+	if !ok {
+		klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("cannot cast runtime object to ClusterApprovalRequest")),
+			"Invalid object type", "object", klog.KObj(obj))
+		return
+	}
+	approved := condition.IsConditionStatusTrue(meta.FindStatusCondition(appReq.Status.Conditions, string(placementv1beta1.ApprovalRequestConditionApproved)), appReq.Generation)
+	approvalAccepted := condition.IsConditionStatusTrue(meta.FindStatusCondition(appReq.Status.Conditions, string(placementv1beta1.ApprovalRequestConditionApprovalAccepted)), appReq.Generation)
+	if approved && approvalAccepted {
+		klog.V(2).InfoS("The approval request has been approved and accepted, ignore queueing for delete event", "clusterApprovalRequest", klog.KObj(appReq))
+		return
+	}
+
+	updateRun := appReq.Spec.TargetUpdateRun
```

Comment:
In the event of updateRun deletion, the controller deletes all associated `clusterApprovalRequests`. Will these deletions trigger updateRun reconcile again? If it's yes, we can ignore this here by checking if updateRun has deletionTimestamp set.

Reply:
Couple of concerns,

- We don't have the client object here to access the updateRun to check deletionTimestamp
- https://github.com/kubefleet-dev/kubefleet/blob/main/pkg/controllers/updaterun/controller.go#L82-L93 this should handle another update run reconcile if it's deleting  right ?